### PR TITLE
feat: GAPIC library BOM in monorepo_script bootstrap

### PR DESCRIPTION
**Pull request by suztomo**
The bootstrap script now creates google-cloud-gapic-bom that covers the GA artifacts released from the monorepo.

### CODE REVIEW

**Review by ddixit14 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -39,3 +39,12 @@ For Day 1 cutover, once we have main branch:
 - Update `googleapis_commit.txt` to an appropriate value
 - Update `.github/workflows/googleapis_hermetic_sync.yaml` to point to
   the main branch.
+
+### The BOM coverage and its version
+
+Review the artifact name "google-cloud-gapic-bom" in the bom directory and
+configure the version managed by Release Please. Ensure the BOM is part of the
+entire release pipeline.
```
Comment (suztomo) [inline README.md:47]: @ddixit14 This message is for you in near future.

Comment (ddixit14) [reply README.md:47]: Sure, will keep this in mind.

File: bom.pom.xml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  <version>0.0.1-SNAPSHOT</version><!-- {x-version-update:google-cloud-gapic-bom:current} -->
+  <name>Google Cloud Java BOM</name>
+  <url>https://github.com/googleapis/java-cloud-bom</url>
+  <description>
+    BOM for the libraries in google-cloud-java repository. Users should not
+    depend on this artifact explicitly because this BOM is an implementation
+    detail of the Libraries BOM.
+  </description>
+  <dependencyManagement>
+BOM_ARTIFACT_LIST
```
Comment (meltsufin) [inline bom.pom.xml:16]: You probably need `<dependencies>` before the placeholder, and a closing one after.

Comment (suztomo) [reply bom.pom.xml:16]: Indeed, the check failed with `Error:  Malformed POM /home/runner/work/google-cloud-java/google-cloud-java/monorepo/google-cloud-java/bom/pom.xml: Unrecognised tag: 'dependency' (position: START_TAG seen ...<dependencyManagement>\n    <dependency>... @16:17)  @ /home/runner/work/google-cloud-java/google-cloud-java/monorepo/google-cloud-java/bom/pom.xml, line 16, column 17`.

File: bootstrap.sh [hunk 3]
```diff
@@ -84,3 +84,42 @@ awk -v MODULE='    <module>CoverageAggregator</module>' '/<\/modules>/{print MOD
...
+
+# generate BOM of the artifacts in this repository
+bom_lines=""
+for bom_directory in $(find . -name 'google-*-bom' | sort); do
+  pom_file="${bom_directory}/pom.xml"
+  groupId_line=$(grep --max-count=1 'groupId' "${pom_file}")
+  artifactId_line=$(grep --max-count=1 'artifactId' "${pom_file}")
+  version_line=$(grep --max-count=1 'x-version-update' "${pom_file}")
+
+  if [[ $version_line == *"<version>0"* ]]; then
```
[the remark this answers is not in the document]
Comment (suztomo) [reply bootstrap.sh:96]: Nice idea.

> why not do this in the main loop above where we iterate over repos.txt?

I feel it's better to split the logic of BOM generation as separate one. No technical reason.

File: bootstrap.sh [hunk 4]
```diff
@@ -84,3 +84,42 @@ awk -v MODULE='    <module>CoverageAggregator</module>' '/<\/modules>/{print MOD
...
+    ${version_line}
+      <type>pom</type>
+      <scope>import</scope>
+    </dependency>
+"
+done
+
+mkdir bom
+awk -v "dependencyManagements=$bom_lines" '{gsub(/BOM_ARTIFACT_LIST/,dependencyManagements)}1' \
+    ../../bom.pom.xml > bom/pom.xml
```
[the remark this answers is not in the document]
Comment (suztomo) [reply bootstrap.sh]: Updated.

**Comment by suztomo** [comment]
`Error:  Non-resolvable import POM: Could not find artifact com.google.cloud:google-identity-accesscontextmanager-bom:pom:1.4.1-SNAPSHOT @ line 24, column 19`

**Comment by suztomo** [comment]
All checks are green.

**Pull request merged**